src/pstl_offload.cpp

```diff
@@ -48,43 +48,111 @@ __get_original_msize()
     return __orig_msize;
 }
 
+struct __hash_aligned_ptr
+{
+    uintptr_t operator()(void *p) const
+    {
+        // we know that addresses are at least page-aligned, so drop 11 right bits that are zeros,
+        // and multiply to some prime number for bits hashing
+        // TODO: should well-known hashing function be used?
+        constexpr unsigned shift = 11;
+        return ((uintptr_t)p >> shift) * 1111111111111111111LLU;
```

Comment:
I guess we can use std::hash<void*> with the shift.

Reply:
Done.